articles/modules/ROOT/pages/how-to-force-direction-of-expansion.adoc

```diff
@@ -0,0 +1,128 @@
+----
+
+There are several different ways we can imagine how this match portion might get planned, subject to indexes present and graph metadata.
+
+If only one of these nodes is chosen as an anchor node, the expansion will start there, expand the relationship to the other node, and then filter the other node's label and properties to find the matches.
+
+If both nodes are looked up via index, still only one will be the anchor node that we expand from, so the expansion process should remain about the same, but filtering will be more efficient, only having to filter on the node's internal graph id to see if the other node is the same one we matched earlier.
+
+It should be clear that there are two possible ways to expand the pattern in this case, and that one is going to be far more efficient than the other.
+
+=== The costly direction
+
+If we anchored on the country node, we would have to expand to every user in the United States, then filter all of them to find the ones that match the pattern.
+This would result in a ton of expanded rows, and a lot of filtering work where we would likely be throwing out every single row, except one.
+
+Even if we find that user early in the filtering, nothing in the query tells it to stop looking (no `LIMIT 1` present), and of course without a unique constraint nothing prevents a node with the same properties from being in the graph (or multiple relationship going to the same node), so it will keep on matching and filtering and throwing out all other non-matching results.
+
+=== The cheap direction
+
+If we anchored to the user node, if we assume that each user only has one `:LIVES_IN` relationship, we would only have to expand on that single relationship and filter on that one connected node to see if the user really does live in the United States.
+
+Even if the graph contains historical data of where a user lives, we can expect only a handful of these paths that we need to filter.
+This approach would still be far quicker and easier than having to filter through every single user in the country to find the path we want.
+
+=== Other expensive pitfalls
+
+There are other ways the planner can decide to plan a query, some of them being cheaper with a small number of nodes and relationships, and some far more costly.
+
+A NodeHashJoin, for example (when we expand from two different anchor nodes to a common node in the middle of the pattern), might be very quick when the number of anchor nodes matched is low, and when we are expanding only a few relationships from the anchor nodes.
+But this can be very expensive if we're traversing a ton of relationships, becoming a hinderance to query execution.
+
+In the case of more complicated queries, there may be quite a few different nodes that could potentially be used as anchor nodes, with many possibilities on which one or which combination to anchor on, and how to expand to fulfill the desired patterns.
```

Comment:
point out to build the query step by step and look at the plan for the number of rows produced from an expansion 
(from either side) to select the better one (and with the knowledge of the model ofc)